virtualmachine/vsphere/vm.go

```diff
@@ -390,6 +391,15 @@ func NewErrorBadResponse(r *http.Response) ErrorBadResponse {
 	return ErrorBadResponse{resp: r}
 }
 
+func isManagedObjectNotFoundError(err error) bool {
```

Comment:
Don't write a method for single object type check. Is it possible to make this method generic so that you could pass the object and type and it can return boolean?